core/divelist.c

```diff
@@ -367,94 +438,128 @@ int get_divesite_idx(struct dive_site *ds)
-		if (pdive->when > dive->when)
-			continue;	/* This could be break if the divelist is always sorted */
+		if (pdive->when >= dive->when) {
+			printf("No - future or same dive\n");
+			break;
+		}
+		/* Don't add the copy of the dive itself */
+		if (i == divenr) {
+			printf("No - copy of dive\n");
+			continue;
+		}
+		printf("Yes\n");
+
 		surface_pressure = get_surface_pressure_in_mbar(pdive, true) / 1000.0;
+		/* Is it the first dive we add? */
 		if (!deco_init) {
+			printf("Init deco\n");
 			clear_deco(surface_pressure);
 			deco_init = true;
-#if DECO_CALC_DEBUG & 2
+			printf("Tissues after init:\n");
 			dump_tissues();
-#endif
 		}
-		if (pdive->when > lasttime) {
-			surface_time = pdive->when - lasttime;
-			lasttime = dive_endtime(pdive);
+		else {
+			surface_time = pdive->when - last_endtime;
+			if (surface_time < 0) {
+				printf("Exit because surface intervall is %d\n", surface_time);
+				return surface_time;
+			}
 			add_segment(surface_pressure, &air, surface_time, 0, dive, prefs.decosac);
-#if DECO_CALC_DEBUG & 2
-			printf("after surface intervall of %d:%02u\n", FRACTION(surface_time, 60));
+			printf("Tissues after surface intervall of %d:%02u:\n", FRACTION(surface_time, 60));
 			dump_tissues();
-#endif
 		}
+
 		add_dive_to_deco(pdive);
-		laststart = pdive->when;
+
+		last_starttime = pdive->when;
+		last_endtime = dive_endtime(pdive);
 		clear_vpmb_state();
-#if DECO_CALC_DEBUG & 2
-		printf("added dive #%d\n", pdive->number);
-		dump_tissues();
-#endif
-	}
-	/* add the final surface time */
-	if (lasttime && dive->when > lasttime) {
-		surface_time = dive->when - lasttime;
-		surface_pressure = get_surface_pressure_in_mbar(dive, true) / 1000.0;
-		add_segment(surface_pressure, &air, surface_time, 0, dive, prefs.decosac);
-#if DECO_CALC_DEBUG & 2
-		printf("after surface intervall of %d:%02u\n", FRACTION(surface_time, 60));
+		printf("Tissues after added dive #%d:\n", pdive->number);
 		dump_tissues();
-#endif
 	}
+
+	surface_pressure = get_surface_pressure_in_mbar(dive, true) / 1000.0;
+	/* We don't have had a previous dive at all? */
 	if (!deco_init) {
-		surface_pressure = get_surface_pressure_in_mbar(dive, true) / 1000.0;
+		printf("Init deco\n");		
 		clear_deco(surface_pressure);
-#if DECO_CALC_DEBUG & 2
-		printf("no previous dive\n");
+		printf("Tissues after no previous dive, surface time set to 48h:\n");
+		dump_tissues();
+	}
+	else {
+		surface_time = dive->when - last_endtime;
+		if (surface_time < 0) {
+			printf("Exit because surface intervall is %d\n", surface_time);
+			return surface_time;
+		}
+		add_segment(surface_pressure, &air, surface_time, 0, dive, prefs.decosac);
+		printf("Tissues after surface intervall of %d:%02u:\n", FRACTION(surface_time, 60));
 		dump_tissues();
-#endif
 	}
+
 	// I do not dare to remove this call. We don't need the result but it might have side effects. Bummer.
-	tissue_tolerance_calc(dive, surface_pressure);
+	//tissue_tolerance_calc(dive, surface_pressure);
```

Comment:
judging from the comment above the `tissue_tolerance_calc(...)` call, can you explain the reasoning behind commenting it out? :-)

this might go in a separate patch, i would say.

Reply:
My mistake! I simply was curious if s.th. is going to explode if I remove this ;-)
BTW: The comment is correct: There is really at least one strange side effect when removing the call so we better keep it.